src/shared/components/plots/BoxScatterPlot.tsx

```diff
@@ -417,6 +417,7 @@ export default class BoxScatterPlot<D extends IBaseBoxScatterPlotPoint> extends
 
     render() {
         if (!this.props.data.length) {
+            setTimeout(()=>this.svgRef(null), 0);
```

Comment:
could you just put this in componentDidUpdate hook?

Reply:
I think I prefer this way for the reason that otherwise we'd have to duplicate the check `if(!this.props.data.length)` and maintain that check if the render logic changes at all. or otherwise create a new variable like "lastRenderWasEmpty" and use that in `componentDidUpdate` to be completely semantically robust.

Reply:
i'm really against this kind of thing when it can be achieved using the normal lifecycle.  what about just checking the data in getter for svgRef?

Reply:
svgRef needs to be triggered exactly when it changes. I'm not sure how to accomplish that cleanly with manual checks. can you explain more?